front/src/applications/operationalStudies/components/MacroEditor/osrdToNge.ts

```diff
@@ -446,18 +452,43 @@ const getNgeLabels = (state: MacroEditorState): LabelDto[] => [
   })),
 ];
 
+export const getTrainrunCategories = (t: TFunction): TrainrunCategory[] => {
+  const mappedCategories: TrainrunCategory[] = Object.entries(OSRD_TRAINRUN_CATEGORY_MAPPING).map(
+    ([key, category]) => ({
+      ...category,
+      name: t(`macroEditor.trainCategory.${key}.name`),
+      shortName: t(`macroEditor.trainCategory.${key}.shortName`),
+    })
+  );
+
+  const noCategory: TrainrunCategory = {
+    id: 10,
+    order: 11,
+    name: t('macroEditor.trainCategory.NO_CATEGORY.name'),
+    shortName: t('macroEditor.trainCategory.NO_CATEGORY.shortName'),
+    fachCategory: 'HaltezeitUncategorized',
+    colorRef: 'NC',
+    minimalTurnaroundTime: 0,
+    nodeHeadwayStop: 0,
+    nodeHeadwayNonStop: 0,
+    sectionHeadway: 0,
+  };
+
+  return [...mappedCategories, noCategory];
+};
+
 /**
  * Return a compatible object for NGE.
  */
-export const getNgeDto = (state: MacroEditorState): NetzgrafikDto => {
+export const getNgeDto = (state: MacroEditorState, t: TFunction<'main'>): NetzgrafikDto => {
   const labels = getNgeLabels(state);
   return {
     ...getNgeTrainrunSectionsWithNodes(state, labels),
     trainruns: getNgeTrainruns(state, labels),
     resources: [state.ngeResource],
     metadata: {
-      netzgrafikColors: [],
-      trainrunCategories: [DEFAULT_TRAINRUN_CATEGORY],
+      netzgrafikColors: getNetzgrafikColors(),
+      trainrunCategories: getTrainrunCategories(t),
       trainrunFrequencies: state.trainrunFrequencies,
```

Comment:
```suggestion
      netzgrafikColors: getNetzgrafikColors(),
      trainrunCategories: state.trainrunCategories,
      trainrunFrequencies: state.trainrunFrequencies,
```

(referencing my other comment) then you can get the categories translated through the state

Reply:
done